### PR TITLE
Do not require DB fields in pdf-map

### PR DESCRIPTION
**Pull request by spokenbird**
[#186402592]

### CODE REVIEW

File: src/main/java/formflow/library/pdf/DatabaseFieldPreparer.java [hunk 1]
```diff
@@ -17,20 +17,23 @@ public class DatabaseFieldPreparer implements DefaultSubmissionFieldPreparer {
...
-        case "flow" -> databaseFields.put(fieldName, new DatabaseField(fieldName, submission.getFlow()));
-        default -> log.error("Unable to map unknown database field: {}", fieldName);
-      }
-    });
+    
+    if (dbFields != null) {
+      dbFields.forEach((fieldName, value) -> {
+        switch (fieldName) {
+          case "submittedAt" ->
+              databaseFields.put(fieldName, new DatabaseField(fieldName, formatDateWithNoTime(submission.getSubmittedAt())));
```
Comment (vrajmohan) [inline src/main/java/formflow/library/pdf/DatabaseFieldPreparer.java:25]: I realize that my comment has nothing to do with _this_ story. Looking around, it feels awkward that the returned `Map` duplicates the `fieldName` in the key and in the `name` property of the value.

File: src/main/java/formflow/library/pdf/DatabaseFieldPreparer.java [hunk 2]
```diff
@@ -17,20 +17,23 @@ public class DatabaseFieldPreparer implements DefaultSubmissionFieldPreparer {
...
+          case "submissionId" -> databaseFields.put(fieldName, new DatabaseField(fieldName, submission.getId().toString()));
+          case "createdAt" ->
+              databaseFields.put(fieldName, new DatabaseField(fieldName, formatDateWithNoTime(submission.getCreatedAt())));
+          case "updatedAt" ->
+              databaseFields.put(fieldName, new DatabaseField(fieldName, formatDateWithNoTime(submission.getUpdatedAt())));
+          case "flow" -> databaseFields.put(fieldName, new DatabaseField(fieldName, submission.getFlow()));
+          default -> log.error("Unable to map unknown database field: {}", fieldName);
```
Comment (vrajmohan) [inline src/main/java/formflow/library/pdf/DatabaseFieldPreparer.java:32]: Do we need this? Do we not "fail early" if there are spurious fields in the YAML?

**Review by vrajmohan — COMMENTED** [review]
Minor comments

File: src/main/java/formflow/library/pdf/DatabaseFieldPreparer.java (hunk 2 above)
Comment (spokenbird) [reply src/main/java/formflow/library/pdf/DatabaseFieldPreparer.java:32]: For this case we do not. We do for the `flows-config` yaml but not for the `pdf-map` yaml. You make a good point though and we probably should be doing that. Maybe we make a ticket to do that outside of this ticket?

**Pull request merged**